src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -5664,7 +5666,7 @@ private static void CombineExtensionMethodArguments(BoundExpression receiver, An
                 hasError = this.CheckInstanceOrStatic(node, receiver, fieldSymbol, ref resultKind, diagnostics);
             }
 
-            if (!hasError && fieldSymbol.IsFixed)
+            if (!hasError && fieldSymbol.IsFixed && !IsBindingNameOf) 
```

Comment:
>IsBindingNameOf [](start = 53, length = 15)

It doesn't look like we need an extra API, all other similar places simply check if EnclosingNameofArgument is not null.